source/api_c/include/c_api.h

```diff
@@ -25,6 +25,19 @@ extern DP_Nlist* DP_NewNlist(int inum_,
                              int* numneigh_,
                              int** firstneigh_);
 
+extern DP_Nlist* DP_NewNlist_comm(int inum_,
+                                  int* ilist_,
+                                  int* numneigh_,
+                                  int** firstneigh_,
+                                  int nswap,
+                                  int* sendnum,
+                                  int* recvnum,
+                                  int* firstrecv,
+                                  int** sendlist,
+                                  int* sendproc,
+                                  int* recvproc,
+                                  void* world);
+
```

Comment:
@coderabbitai please generate function documentation according to `source/lib/include/neighbor_list.h`.

Reply:
Generate suggestion.

Reply:
@CaRoLZhangxy commit it if it is okay